cmake/modules/FindMicroHttpd.cmake

```diff
@@ -35,16 +35,8 @@ if(MICROHTTPD_FOUND)
   set(MICROHTTPD_INCLUDE_DIRS ${MICROHTTPD_INCLUDE_DIR})
   set(MICROHTTPD_DEFINITIONS -DHAS_WEB_SERVER=1 -DHAS_WEB_INTERFACE=1)
 
-  if(KODI_DEPENDSBUILD AND NOT WIN32)
-    find_library(GCRYPT_LIBRARY gcrypt)
-    find_library(GPGERROR_LIBRARY gpg-error)
-    list(APPEND MICROHTTPD_LIBRARIES ${GCRYPT_LIBRARY} ${GPGERROR_LIBRARY})
-    mark_as_advanced(GCRYPT_LIBRARY GPGERROR_LIBRARY)
-    if(NOT APPLE AND NOT CORE_SYSTEM_NAME STREQUAL android)
-      list(APPEND MICROHTTPD_LIBRARIES rt)
-    endif()
-  else()
-    list(APPEND MICROHTTPD_LIBRARIES ${PC_MICROHTTPD_STATIC_LIBRARIES})
+  if(${MICROHTTPD_LIBRARY} MATCHES ".+\.a$" AND PC_MICROHTTPD_STATIC_LDFLAGS)
+    list(APPEND MICROHTTPD_LIBRARIES ${PC_MICROHTTPD_STATIC_LDFLAGS})
```

Comment:
This still lists unistring, so it results in essentially the same behavior as without this PR for depends builds.

Reply:
This is not for fixing depends builds, it's for fixing builds with shared system libs.

Reply:
Now I'm confused, since I'm not aware that they are broken. PPA builds have been fine for a long time.
I also took another look and even though unistring is referenced in the .pc file, cmake seems to filter this out:
>PC_MICROHTTPD_STATIC_LDFLAGS:INTERNAL=-lmicrohttpd;-lgnutls;-lgmp;/usr/lib/x86_64-linux-gnu/libunistring.so;-lidn2;-lhogweed;-lgmp;-lnettle;-ltasn1;-lp11-kit;-lz
PC_MICROHTTPD_STATIC_LDFLAGS_OTHER:INTERNAL=/usr/lib/x86_64-linux-gnu/libunistring.so
**PC_MICROHTTPD_STATIC_LIBRARIES:INTERNAL=microhttpd;gnutls;gmp;idn2;hogweed;gmp;nettle;tasn1;p11-kit;z**

Reply:
### system libs
```
+ pkg-config --libs libmicrohttpd
-lmicrohttpd
+ pkg-config --libs-only-l libmicrohttpd
-lmicrohttpd
+ pkg-config --libs-only-L libmicrohttpd

+ pkg-config --libs-only-other libmicrohttpd

+ pkg-config --libs --static libmicrohttpd
-lmicrohttpd -lgnutls -lgmp -lunistring -lidn2 -latomic -lhogweed -lgmp -lnettle -ltasn1 -lp11-kit
+ pkg-config --libs-only-l --static libmicrohttpd
-lmicrohttpd -lgnutls -lgmp -lunistring -lidn2 -latomic -lhogweed -lgmp -lnettle -ltasn1 -lp11-kit
+ pkg-config --libs-only-L --static libmicrohttpd

+ pkg-config --libs-only-other --static libmicrohttpd

```
```
PC_MICROHTTPD_STATIC_LDFLAGS:INTERNAL=-lmicrohttpd;-lgnutls;-lgmp;-lunistring;-lidn2;-latomic;-lhogweed;-lgmp;-lnettle;-ltasn1;-lp11-kit
PC_MICROHTTPD_STATIC_LIBRARIES:INTERNAL=microhttpd;gnutls;gmp;unistring;idn2;atomic;hogweed;gmp;nettle;tasn1;p11-kit
```

### depends
```
+ pkg-config --libs libmicrohttpd
-L<PREFIX>lib -lmicrohttpd
+ pkg-config --libs-only-l libmicrohttpd
-lmicrohttpd
+ pkg-config --libs-only-L libmicrohttpd
-L<PREFIX>lib
+ pkg-config --libs-only-other libmicrohttpd

+ pkg-config --libs --static libmicrohttpd
-L<PREFIX>lib -lmicrohttpd -lgcrypt -lgpg-error -lgnutls -lz -lgmp -lhogweed -lgmp -lnettle
+ pkg-config --libs-only-l --static libmicrohttpd
-lmicrohttpd -lgcrypt -lgpg-error -lgnutls -lz -lgmp -lhogweed -lgmp -lnettle
+ pkg-config --libs-only-L --static libmicrohttpd
-L<PREFIX>lib
+ pkg-config --libs-only-other --static libmicrohttpd

```

I don't know what's different on your system, but linking with `PC_MICROHTTPD_STATIC_LIBRARIES` if microhttpd is a shared library is always wrong.

Reply:
I agree it's unexpected, but the thing is, it's not only my system, it also works fine on launchpad. And since launchpad uses completely clean ubuntu, this must be something on your end.
I'd like to find out what.

But using PC_MICROHTTPD_STATIC_LDFLAGS breaks it for me, because now unistring is not filtered anymore

Reply:
If you are using the *shared* system library `PC_MICROHTTPD_STATIC_LDFLAGS` won't be added to `MICROHTTPD_LIBRARIES`, because of the `.a` check you requested to add. If you want, I can replace `PC_MICROHTTPD_STATIC_LDFLAGS` with `PC_MICROHTTPD_STATIC_LIBRARIES`.

Reply:
yes, plz change to PC_MICROHTTPD_STATIC_LIBRARIES

Reply:
done